calendar-app/src/pages/calendar/Event.tsx

```diff
@@ -0,0 +1,215 @@
+import saveEvent from "../../api/saveAndEditEvent";
+import deleteEvent from "../../api/deleteEvent";
+import { useDispatch } from "react-redux";
+import { ChangeEvent, useEffect, useState } from "react";
+import { StoredEvent } from "../../utils/types";
+import { removeEvent } from "../../features/eventDisplay";
+
+export class Form implements StoredEvent {
+  id = 0;
+  title = "";
+  startTime = "";
+  startDate = "";
+  endTime = "";
+  endDate = "";
+  guests? = "";
+  location? = "";
+  description? = "";
+  [key: string]: string | number | undefined;
+  eventKey = "";
+}
+
+function Event({
+  eventWindow,
+  triggerEventWindow,
+  initialEvent,
+}: {
+  eventWindow: boolean;
+  triggerEventWindow: (value: boolean) => void;
+  initialEvent?: StoredEvent | null;
+}) {
+  const dispatch = useDispatch();
+  const [form, setForm] = useState<Form>(new Form());
```

Comment:
react state should usually be a simple object (serialisable), not an object of a class (with methods, inner properties etc.)